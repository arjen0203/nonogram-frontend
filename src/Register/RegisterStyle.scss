@import './../colors';

.center {
    justify-content: center;
    display: flex;

    .register-fields {
        display: flex;
        flex-direction: column;
        background-color: $blackCoral;
        padding: 20px;
        border-radius: 20px;
        margin-top: 60px;
        width: 350px;

        * {
            color: $white;
            font-size: 18px;
        }

        .register-title {
            text-align: center;
            color: $white;
        }

        .register-username-input {
            margin-top: 3px;
            margin-bottom: 10px;
            color: $blackCoral;
            padding-left: 3px;
        }

        .register-password-input {
            margin-top: 3px;
            margin-bottom: 10px;
            color: $blackCoral;
            padding-left: 3px;
        }

        .register-error {
            color: $errorColorLight;
            font-size: 20px;
            text-align: center;
            margin-top: 8px;
            margin-bottom: -8px;
        }

        .register-submit-button {
            display: inline-block;
            margin-top: 5px;
            padding: 5px;
            background-color: $silverSand;
            color: white;
            cursor: pointer;
            text-decoration: none;
            border: none;
            transition: 0.4s;
            width: 200px;
            margin-left: 75px;
            color: $gunMetal;
            border-radius: 5px;

            &:hover {
                background-color: $silverSandLight;
            }
            &:active {
                position:relative;
                top:1px;
            }
        }
        .to-login-link {
            text-align: center;
            margin-top: 10px;
        }
    }
}
