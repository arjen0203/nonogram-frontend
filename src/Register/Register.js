import React, {Component} from 'react';
import {Link} from "react-router-dom";
import './RegisterStyle.scss';

class Register extends Component {
    constructor(props) {
        super(props);

        this.state = {
            username: "",
            password: "",
            passwordRepeat: ""
        }

        this.handleNameChange = this.handleNameChange.bind(this);
        this.handlePasswordChange = this.handlePasswordChange.bind(this);
        this.handlePasswordRepChange = this.handlePasswordRepChange.bind(this);
        this.tryRegistrating = this.tryRegistrating.bind(this);
    }

    tryRegistrating(){
        this.saving = true;

        let user = { username: this.state.username, password: this.state.password };
        console.log(JSON.stringify(user));
        fetch(`http://localhost:8080/api/user/add`, {
            method: 'POST',
            headers: { 'Content-Type': 'application/json' },
            body: JSON.stringify(user)
        })
            .then(res => {
                if(!res.ok){
                    this.setState({ ...this.state, registerError: "Something went wrong while registrating"});
                    this.saving = false;
                    return Promise.reject("Promise rejected")
                }
                //res.json()
            })
            .then(data => {
                this.props.history.push("/");
                this.saving = false;
            })
            .catch(this.setState({ ...this.state, registerError: "Something went wrong while registrating"}));
    }

    handleNameChange(event){
        this.setState({username: event.target.value});
    }

    handlePasswordChange(event){
        this.setState({password: event.target.value});
    }
    handlePasswordRepChange(event){
        this.setState({passwordRepeat: event.target.value});
    }

    render() {
        return (
            <div className="center">
                <div className="register-fields">
                    <b className="register-title">Register:</b>
                    <label>Username:</label>
                    <input className="register-username-input" type="text" placeholder=" Username" value={this.state.username} onChange={this.handleNameChange}></input>
                    <label>Password:</label>
                    <input className="register-password-input" type="password" placeholder=" Password" value={this.state.password} onChange={this.handlePasswordChange}></input>
                    <label>Password repeat:</label>
                    <input className="register-password-input" type="password" placeholder=" Password repeat" value={this.state.passwordRepeat} onChange={this.handlePasswordRepChange}></input>
                    <button className="register-submit-button" onClick={this.tryRegistrating}>Register</button>

                    <Link className="to-login-link" to={'/login'}>Already have an account? Login here.</Link>
                </div>
            </div>
        );
    }
}

export default Register;