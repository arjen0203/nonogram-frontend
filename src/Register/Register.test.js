import React from "react";
import ReactDOM from 'react-dom';
import Register from "./Register";
import {render, cleanup} from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect"

import renderer from "react-test-renderer";

afterEach(cleanup);

it('Renders without crashing', function () {
	const div = document.createElement('div');
	ReactDOM.render(<Register></Register>, div);
});

it('matches snapshot', function () {
	const tree = renderer.create(<Register></Register>).toJSON();
	expect(tree).toMatchSnapshot();
});