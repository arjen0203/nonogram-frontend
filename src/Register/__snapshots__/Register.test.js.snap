// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="center"
>
  <div
    className="register-fields"
  >
    <b
      className="register-title"
    >
      Register:
    </b>
    <label
      htmlFor="username"
    >
      Username:
    </label>
    <input
      className="register-username-input"
      id="username"
      onChange={[Function]}
      placeholder="Username"
      type="text"
      value=""
    />
    <label
      htmlFor="password"
    >
      Password:
    </label>
    <input
      className="register-password-input"
      id="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
    <label
      htmlFor="passwordRep"
    >
      Password repeat:
    </label>
    <input
      className="register-password-input"
      id="passwordRep"
      onChange={[Function]}
      placeholder="Password repeat"
      type="password"
      value=""
    />
    <button
      className="register-submit-button"
      onClick={[Function]}
    >
      Register
    </button>
    <b
      className="register-error"
    />
    <div
      className="to-login-link"
      onClick={[Function]}
    >
      Already have an account? Login here.
    </div>
  </div>
</div>
`;
